File: AppliSportSpringBoot/src/main/java/projet/AppliSport/model/Sport.java
package projet.AppliSport.model;

import java.util.Objects;
import java.util.Set;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.ForeignKey;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.OneToMany;
import javax.persistence.OneToOne;
import javax.persistence.SequenceGenerator;
import javax.persistence.Version;
import javax.validation.constraints.NotEmpty;

import com.fasterxml.jackson.annotation.JsonView;

import projet.AppliSport.views.Views;
import projet.AppliSport.views.Views.SportWithClub;

@Entity
@SequenceGenerator(name = "seqSport", sequenceName = "seq_sport", initialValue = 10, allocationSize = 1)
public class Sport {
	@Id
	@GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "seqSport")
	@JsonView(Views.Common.class)
	private Long id;
	
	@NotEmpty
	@Column(name = "nom", length = 150)
	@JsonView(Views.Common.class)
	private String nom;
	
	@OneToMany(mappedBy = "sportClub")
	@JsonView(Views.SportWithClub.class)
	private Set<Club> clubs;
	@OneToMany(mappedBy = "sport")
	@JsonView(Views.SportWithInteret.class)
	private Set<Interet> interets;
	
	@OneToOne
	@JoinColumn(name = "sport_caracteristique", foreignKey = @ForeignKey(name = "sport_caracteristique_fk"))
	@JsonView(Views.SportWithCaracteristique.class)
	private Caracteristique caracteristique;

	@Version
	private int version;
	
	public Sport() {
		
	}
	
	




	public String getNom() {
		return nom;
	}



	public void setNom(String nom) {
		this.nom = nom;
	}



	
	
	
	public Long getId() {
		return id;
	}

	public void setId(Long id) {
		this.id = id;
	}

	public Set<Club> getClubs() {
		return clubs;
	}

	public void setClubs(Set<Club> clubs) {
		this.clubs = clubs;
	}

	public Set<Interet> getInterets() {
		return interets;
	}






	public void setInterets(Set<Interet> interets) {
		this.interets = interets;
	}


	




	public Caracteristique getCaracteristique() {
		return caracteristique;
	}






	public void setCaracteristique(Caracteristique caracteristique) {
		this.caracteristique = caracteristique;
	}






	public int getVersion() {
		return version;
	}






	public void setVersion(int version) {
		this.version = version;
	}






	@Override
	public int hashCode() {
		return Objects.hash(id);
	}


	@Override
	public boolean equals(Object obj) {
		if (this == obj)
			return true;
		if (obj == null)
			return false;
		if (getClass() != obj.getClass())
			return false;
		Sport other = (Sport) obj;
		return Objects.equals(id, other.id);
	}


	
	
}

File: SportifyApp/src/app/model/sport.ts
import { Caracteristique } from './caracteristique';
import { Club } from './club';
import { Interet } from './interet';

export class Sport {
  private _id: number | undefined;
  private _nom: string | undefined;
  private _clubs: Club[] | undefined;
  private _interets: Interet[] | undefined;
  private _caracteristique: Caracteristique | undefined;

  constructor(
    id?: number,
    nom?: string,
    clubs?: Club[],
    interets?: Interet[],
    caracteristique?: Caracteristique
  ) {
    this._id = id;
    this._nom = nom;
    this._clubs = clubs;
    this._interets = interets;
    this._caracteristique = caracteristique;
  }

  /**
   * Getter id
   * @return {number }
   */
  public get id(): number | undefined {
    return this._id;
  }

  /**
   * Getter nom
   * @return {string }
   */
  public get nom(): string | undefined {
    return this._nom;
  }

  /**
   * Getter clubs
   * @return {Club[] }
   */
  public get clubs(): Club[] | undefined {
    return this._clubs;
  }

  /**
   * Getter interets
   * @return {Interet[] }
   */
  public get interets(): Interet[] | undefined {
    return this._interets;
  }

  /**
   * Setter id
   * @param {number } value
   */
  public set id(value: number | undefined) {
    this._id = value;
  }

  /**
   * Setter nom
   * @param {string } value
   */
  public set nom(value: string | undefined) {
    this._nom = value;
  }

  /**
   * Setter clubs
   * @param {Club[] } value
   */
  public set clubs(value: Club[] | undefined) {
    this._clubs = value;
  }

  /**
   * Setter interets
   * @param {Interet[] } value
   */
  public set interets(value: Interet[] | undefined) {
    this._interets = value;
  }

  /**
   * Getter caracteristique
   * @return {Caracteristique }
   */
  public get caracteristique(): Caracteristique | undefined {
    return this._caracteristique;
  }

  /**
   * Setter caracteristique
   * @param {Caracteristique } value
   */
  public set caracteristique(value: Caracteristique | undefined) {
    this._caracteristique = value;
  }
}
